Clarify what/why/how (#65)
JuliaRegistries/TagBot: README.md

When we talk about tags and releases, we mean *Git tags* and *GitHub releases*, and not releases in a registry that allow the Julia package manager to install your package.
TagBot does not register your package for you, see the documentation in [General](https://github.com/JuliaRegistries/General/blob/master/README.md) and [Registrator](https://github.com/JuliaRegistries/Registrator.jl/blob/master/README.md) for that.
Instead, it reacts to versions of your packages that have been registered, making TagBot a set-and-forget solution to keep your repository in sync with your package releases. Tags and releases aren't at all necessary, but it's considered a good practice.

> ### Benefits of using TagBot

Reviewer: I've put the contents of this section above into the summary, I don't think it needs its own section.

> - Manual tagging and releasing on your Github repo is no longer necessary.

Reviewer: This is not really a benefit because Git tags and GitHub releases are not necessary at all -- whether they're manual or automated. I've replaced it above with another benefit: "Users can easily browse your package code at specific releases."

- TagBot will automatically include release notes on the GitHub release page

> - Lots of configuration options (e.g. use a registry other than the General Registry)

Reviewer: I don't usually consider the configuration options of a tool to be a benefit of that tool except when comparing it to other tools that do the same thing. It's kind of like answering "why should I use version control?" with "because Git has a lot of configuration options!".

          token: ${{ secrets.GITHUB_TOKEN }}
```

> No further action is required on your part. When you register a new release with the Julia General Registry (either manually or with [Registrator](https://github.com/JuliaRegistries/Registrator.jl/blob/master/README.md)), TagBot will create a Github Tag and Github Release on your package's Github repository.  

Reviewer: ```suggestion
No further action is required on your part.
When you add a new release to a registry with Registrator, TagBot will create a GitHub release on your package's repository.
```

Just being a bit less verbose:
- Releases aren't necessarily in General
- I don't think anyone is manually editing the registry
- We already linked to Registrator above
- Most of the time, TagBot will create lightweight tags so just mentioning that it creates a GitHub release is sort of more accurate

> You may, however, want to customize the behavior via the available configuration options:

Reviewer: ```suggestion
You may, however, want to customize the behaviour via the available configuration options:
```

Proper English please :wink:
